common/math/math.go

```diff
@@ -34,3 +46,195 @@ func RoundFloat(x float64, prec int) float64 {
+// usually an index, compared to the volatility of those returns.
+// The benchmark used is typically an index that represents the market or a particular sector or industry.
+func InformationRatio(values, benchmarkRates []float64, averageValues, averageComparison float64) (float64, error) {
+	if len(benchmarkRates) == 1 {
+		for i := range values {
+			if i == 0 {
+				continue
+			}
+			benchmarkRates = append(benchmarkRates, benchmarkRates[0])
+		}
+	}
+	var diffs []float64
+	for i := range values {
+		diffs = append(diffs, values[i]-benchmarkRates[i])
+	}
+	stdDev, err := PopulationStandardDeviation(diffs)
+	if err != nil {
+		return 0, err
+	}
+	if stdDev == 0 {
+		return 0, nil
+	}
+	return (averageValues - averageComparison) / stdDev, nil
+}
+
+// PopulationStandardDeviation calculates standard deviation using population based calculation
+func PopulationStandardDeviation(values []float64) (float64, error) {
+	if len(values) < 2 {
+		return 0, nil
+	}
+	valAvg, err := ArithmeticAverage(values)
+	if err != nil {
+		return 0, err
+	}
+	diffs := make([]float64, len(values))
+	for x := range values {
+		diffs[x] = math.Pow(values[x]-valAvg, 2)
+	}
+	var diffAvg float64
+	diffAvg, err = ArithmeticAverage(diffs)
+	if err != nil {
+		return 0, err
+	}
+	return math.Sqrt(diffAvg), nil
+}
+
+// SampleStandardDeviation standard deviation is a statistic that
+// measures the dispersion of a dataset relative to its mean and
+// is calculated as the square root of the variance
+func SampleStandardDeviation(vals []float64) (float64, error) {
+	if len(vals) < 2 {
+		return 0, nil
+	}
+	mean, err := ArithmeticAverage(vals)
+	if err != nil {
+		return 0, err
+	}
+	var superMean []float64
+	var combined float64
+	for i := range vals {
+		result := math.Pow(vals[i]-mean, 2)
+		superMean = append(superMean, result)
+		combined += result
+	}
+	avg := combined / (float64(len(superMean)) - 1)
+	return math.Sqrt(avg), nil
+}
+
+// GeometricAverage is an average which indicates the central tendency or
+// typical value of a set of numbers by using the product of their values
+// The geometric average can only process positive numbers
+func GeometricAverage(values []float64) (float64, error) {
+	if len(values) == 0 {
+		return 0, errZeroValue
+	}
+	product := 1.0
+	for i := range values {
+		if values[i] <= 0 {
+			// cannot use negative or zero values in geometric calculation
+			return 0, errGeometricNegative
+		}
+		product *= values[i]
+	}
+	geometricPower := math.Pow(product, 1/float64(len(values)))
+	return geometricPower, nil
+}
+
+// FinancialGeometricAverage is a modified geometric average to assess
+// the negative returns of investments. It accepts It adds +1 to each
+// This does impact the final figures as it is modifying values
+// It is still ultimately calculating a geometric average
+// which should only be compared to other financial geometric averages
+func FinancialGeometricAverage(values []float64) (float64, error) {
```

Comment:
The name should instead be FinancialGeometricMean. Even though average and mean have the same formula they still have slightly different meanings. Changing this to mean will also keep it consistent with the error msgs since you have put 'geometric mean' in the error msgs